test/th_tr_mixed.jl

```diff
@@ -0,0 +1,44 @@
+using ThArrays
+using Tracker: forward, data
+using Random
+using Test
+
+using ThArrays.TrackerAD: _th, _tr
+
+@testset "Torch and Tracker Mixed Gradient" begin
+
+    Random.seed!(0);
+
+    a = rand(3, 2)
+    b = rand(3, 2)
+
+    @testset "Simple Mixed AD" begin
+        # all with th(PyTorch Backend)
+        f1(x, y) = sum(sin.(_th(x)) + sin.(_th(y)))
```

Comment:
Maybe improve the API by selecting AD backend based on the type of `x` and `y`, for example:

```julia
# sin.(y) with th, everything else with tr
f1(x::TrackedArray, y::Tensor) = sum(sin.(x) + sin.(y))
```
In general, functions are dispatched based on argument types (Tensor or TrackedArray). When mixed types are passed to a function, `Tensors` are automatically promoted to `TrackArray` by default (or the other way).

Reply:
This way have two problems:
- We can't tell in which way the user would like to track the `+` operation, and we have more nested operators in the function body, we have no way to switch them forth and back.
- Generally, we just pass Arrays to `f1`, passing `TrackedArray` or `Tensor` may not be such convenient.